tests/Template/KeyOfTemplateTest.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psalm\Tests;
+
+use Psalm\Tests\Traits\InvalidCodeAnalysisTestTrait;
+use Psalm\Tests\Traits\ValidCodeAnalysisTestTrait;
+
+class KeyOfTemplateTest extends TestCase
+{
+    use InvalidCodeAnalysisTestTrait;
+    use ValidCodeAnalysisTestTrait;
+
+    /**
+     * @return iterable<string,array{code:string,assertions?:array<string,string>,ignored_issues?:list<string>,php_version?:string}>
+     */
+    public function providerValidCodeParse(): iterable
+    {
+        return [
+            'acceptsArrayKeysFn' => [
+                'code' => '<?php
+                    /**
+                     * @template T of array
+                     * @param T $array
+                     * @return key-of<T>[]
+                     */
+                    function getKey($array) {
+                        return array_keys($array);
+                    }
+                '
+            ],
+            'acceptsArrayKeyFirstFn' => [
+                'code' => '<?php
+                    /**
+                     * @template T of array
+                     * @param T $array
+                     * @return key-of<T>|null
+                     */
+                    function getKey($array) {
+                        return array_key_first($array);
+                    }
+                '
+            ],
+            'acceptsArrayKeyLastFn' => [
+                'code' => '<?php
+                    /**
+                     * @template T of array
+                     * @param T $array
+                     * @return key-of<T>|null
+                     */
+                    function getKey($array) {
+                        return array_key_last($array);
+                    }
+                '
+            ],
+            // Currently not works!
+            // 'acceptsIfArrayKeyExistsFn' => [
```

Comment:
Would you mind uncommenting and putting `SKIPPED-` in front of the name? Most of the other failing tests due to missing features are skipped that way, and it makes it much easier to later un-skip all of the skipped tests to see if any have started passing.

Reply:
Done